## Stats Assignment 3

Problem Statement 1: 
    
Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on  blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet  have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an  effect or not. 


In [2]:
import numpy as np

𝐻0: 𝜇 = 100<br/>
𝐻1: 𝜇 ≠ 100

In [3]:
mu = 100
std = 15
n = 36
x = 108
alpha = 0.05

z_value =  1.96 # 1 - (0.05/2) = 0.975 ; 0.975 value from Z test table we get : 1.9 + 0.06 = 1.96

z_score = (x - mu)/(std/np.sqrt(n))

if z_score > -z_value and z_score < z_value:
    print(f'Null Hypothesis(H0) is accepted as value of Z Score lies between {-z_value,z_value}')
    print(f'Value of Z Score is: {z_score:.2f} ')
else :
    print('Null Hypothesis(H0) is rejected')
    print(f'Value of Z Score is: {z_score:.2f} ')

Null Hypothesis(H0) is rejected
Value of Z Score is: 3.20 


Problem Statement 2: 

In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second  state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple  random sample of 100 voters are surveyed from each state. 
What is the probability that the survey will show a greater percentage of Republican  voters in the second state than in the first state?


In [9]:
#P1 = the proportion of Republican voters in the first state
#P2 = the proportion of Republican voters in the second state
#p1 = the proportion of Republican voters in the sample from the first state
#p2 = the proportion of Republican voters in the sample from the second state. 
#n1 = the number of voters sampled from the first state 
#n2 = the number of voters sampled from the second state  

P1 = 0.52
P2 = 0.48
n1 = 100
n2 = 100

#The mean of the difference in sample proportions .i.e Expected Value E[p1 -p2] = P1 - P2 = mu
mu =  P1 - P2

#the standard deviation of the difference
sigma = np.sqrt((((P1 * (1-P1) ) / n1) + ((P2 * (1-P2)) /n2)))

#This problem requires us to find the probability that p1 is less than p2. 
#This is equivalent to finding the probability that p1 - p2 < 0. 
x = 0
#To find this probability, we need to transform the random variable (p1 - p2) into a z-score. 
#That transformation appears below.
Z_p1_p2 = (x - mu)/sigma

print("Z_score(p1,p2) : ",Z_p1_p2)

Z_score(p1,p2) :  -0.5661385170722983


From Z table we find that the probability of a z-score being -0.5661 or less is 0.24.

Therefore, the probability that the survey will show a greater percentage of Republican voters 
in the second state than in the first state is 0.24.

Problem Statement 3: 
    
You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard  deviation is 209. How well did you score on the test compared to the average test taker? 


In [11]:
x = 1100 
mu = 1026 
std = 209 
z = (x - mu)/std
print("Z Score : ",z)

Z Score :  0.35406698564593303


The above calculation shows that my score is 0.35 standard deviations above the mean(average) test score